org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/PasteEventHandler.java

```diff
@@ -334,6 +345,75 @@ private static String getEol(String text) {
 		return text.contains("\r\n") ? "\r\n" : "\n";
 	}
 
+	public static String handleFilePasteEvent(String path, String content, IProgressMonitor monitor) throws JavaModelException {
+		String desiredPath = "";
+		final ASTParser parser = ASTParser.newParser(AST.getJLSLatest());
+		parser.setResolveBindings(false);
+		parser.setKind(ASTParser.K_COMPILATION_UNIT);
+		parser.setBindingsRecovery(false);
+		parser.setIgnoreMethodBodies(true);
+		Map<String, String> javaOptions = JavaCore.getOptions();
+		parser.setCompilerOptions(javaOptions);
+		parser.setSource(content.toCharArray());
+		CompilationUnit root = (CompilationUnit) parser.createAST(monitor);
+		// Check for package, import, and/or type declaration to determine if java file
+		if (root != null && (root.getPackage() != null || !root.imports().isEmpty() || !root.types().isEmpty())) {
+			String pathRoot = path;
+			if (root.getPackage() != null) {
+				String packageName = root.getPackage().getName().getFullyQualifiedName();
+				IJavaProject[] projects = ProjectUtils.getJavaProjects();
+				boolean pathFound = false;
+				String partialPathAddition = "";
```

Comment:
I think you should break this loop out into its own method that returns an IResource when it finds the best one. 

Even with your attempt to check if pathFound in your if-statement, it doesn't end the loop. You have two nested loops here, and the break command won't let escape both of them. Having it in its own method that returns when it finds what it wants would be better.